new-log-viewer/src/contexts/UrlContextProvider.tsx

```diff
@@ -0,0 +1,241 @@
+import React, {
+    createContext,
+    useEffect,
+    useState,
+} from "react";
+
+import {NullableProperties} from "../typings/common";
+import {
+    HASH_PARAM_NAME,
+    SEARCH_PARAM_NAME,
+    UrlHashParams,
+    UrlHashParamUpdatesType,
+    UrlParamsType,
+    UrlSearchParams,
+    UrlSearchParamUpdatesType,
+} from "../typings/url";
+import {getAbsoluteUrl} from "../utils/url";
+
+
+const UrlContext = createContext <UrlParamsType>({} as UrlParamsType);
+
+/**
+ * Default values of the search parameters.
+ */
+const URL_SEARCH_PARAMS_DEFAULT = Object.freeze({
+    [SEARCH_PARAM_NAME.FILE_PATH]: null,
+});
+
+/**
+ * Default values of the hash parameters.
+ */
+const URL_HASH_PARAMS_DEFAULT = Object.freeze({
+    [HASH_PARAM_NAME.LOG_EVENT_NUM]: null,
+});
+
+/**
+ * Computes updated URL search parameters based on the provided key-value pairs.
+ *
+ * @param updates An object containing key-value pairs to update the search parameters.
+ * If a key's value is `null`, the key will be removed from the search parameters.
+ * @return The updated search parameters string.
+ */
+const getUpdatedSearchParams = (updates: UrlSearchParamUpdatesType) => {
+    const newSearchParams = new URLSearchParams(window.location.search.substring(1));
+    const {filePath} = updates;
+
+    for (const [key, value] of Object.entries(updates)) {
+        if (SEARCH_PARAM_NAME.FILE_PATH as string === key) {
+            continue;
+        }
+        if (null === value) {
+            newSearchParams.delete(key);
+        } else {
+            newSearchParams.set(key, String(value));
+        }
+    }
+    if (null === filePath) {
+        newSearchParams.delete(SEARCH_PARAM_NAME.FILE_PATH);
+    } else if ("string" === typeof filePath) {
+        newSearchParams.set(SEARCH_PARAM_NAME.FILE_PATH, filePath);
```

Comment:
Why do we need this check?

Reply:
Previously parameter `updates` wasn't typed and `filePath` could be of other types (undefined or non-string). I just updated the check to be more explicit.